ospfd/ospf_snmp.c

```diff
@@ -641,7 +641,7 @@ static struct ospf_area *ospfAreaLookup(struct variable *v, oid name[],
 
 	if (exact) {
 		/* Length is insufficient to lookup OSPF area. */
-		if (*length - v->namelen != sizeof(struct in_addr))
+		if (*length != (size_t)v->namelen + IN_ADDR_SIZE)
```

Comment:
please just change the IN_ADDR_SIZE and IN6_ADDR_SIZE in smux.h to have a UL

Reply:
It was unsigned long before https://github.com/FRRouting/frr/commit/7f1b2a509e1756a07b75d10d621dd27447a29068